### ISSUE TITLE
Issue with valid OTF fonts not being converted properly.

### ISSUE THREAD

**Issue by onassar** [rillig/sfntly#3]
I believe this is a new issue because I used a library that build sfntly from the original repo, and there aren't any issues. These fonts don't work properly:
https://expirebox.com/download/289aaab245dbfa1408f5cd5b899b6b88.html

**Comment by rillig** [comment]
Please describe your issue more precisely:

* What does "not being converted properly" mean?
* What kind of conversion do you mean?
* Which exact steps did you do?
* What result did you expect?
* What was the actual outcome?

I ran the `EOTWriter` with the Open Sans font that is part of the repo, and both my current and the original version produce exactly the same file.

**Comment by onassar** [comment]
I was told to post here, as I ran into an issue with a library that uses your repo:
https://github.com/bdusell/webfont-generator/issues/15

**Comment by rillig** [comment]
Ah, that's much better than just "sfntly doesn't work". I'll have a look at it. Thanks for reporting it.

**Comment by onassar** [comment]
I didn't write "sfntly doesn't work"
I gave two examples of fonts that would not convert.

Let me know if you need anything else to debug.

**Comment by rillig** [comment]
I cannot reproduce the problem completely.

I also get the `OutOfMemoryError` when running this command:

    ['java', '-Xmx4g', '-cp', './src/java;./vendor/sfntly/java/build/classes', 'ConvertFont', 'test-out/otf.svg', '-o', 'test-out/otf.eot', '-o', 'test-out/otf.woff']

(I'm working on Windows, therefore I had to replace the absolute cygwin paths with the dot and the `:` separator with a `;`.)

To verify whether I introduced the bug with my cleanup commits, I rolled back the sfntly repository to commit ed46801c from February, which is long before I started working on sfntly. I could reproduce the `OutOfMemoryError` even with that version.

To help me nail down the cause, you could tell me a "good" and a "bad" configuration of the webfont-generator projects. What do I need to checkout so that it works again? Then I can hunt down the actual bug.

**Comment by onassar** [comment]
That's a little out of my experience/knowledge, but I can point you to the tagged version of the repo that was working: https://github.com/bdusell/webfont-generator/tree/8b1a502855cd755ce6e768bc5e340d36421ceeb6

**Comment by rillig** [comment]
Thanks for the working commit, this already helps me. I'm tracking it down right now.

**Comment by onassar** [comment]
Great. Happy to help however I can. I'll ping @bdusell once you push up a fix to integrate it into his library, and then am happy to test it.

**Comment by rillig** [comment]
The tagged version that you gave me doesn't work. It also results in an `OutOfMemoryError`. Here is what I did:

```shell
cd
cd git
git clone git@github.com:bdusell/webfont-generator.git wegood
cd wegood
./setup
./bin/generate-webfonts  -o test-out/ --css test.css ~/onassar-fonts/otf.otf
```

If you have any better setup steps, please share them with me.

**Comment by onassar** [comment]
Looks like you're cloning master. Try download that specific hash and building from that.
https://github.com/bdusell/webfont-generator/tree/8b1a502855cd755ce6e768bc5e340d36421ceeb6

**Comment by rillig** [comment]
I finally found the problem. The Java command line that is generated by the webfonts-generator is:

    ['java', '-cp', './src/java;./vendor/sfntly/java/target/classes', 'ConvertFont', 'test-out/otf.svg', '-o', 'test-out/otf.eot', '-o', 'test-out/otf.woff']

This means that the webfonts-generator asks sfntly to read an SVG font file. But sfntly cannot do this. (To be honest, it should provide a much nicer error message.)

Sfntly can only read OTF and TTF files, not SVG. Therefore the webfonts-generator must be fixed.

**Comment by onassar** [comment]
Oh interesting okay. @bdusell does that make sense to you?

**Comment by rillig** [comment]
In parallel, I will change sfntly so that it checks whether the provided file is in the correct format. The current state of "just believing untrusted data" is clearly wrong.

**Comment by onassar** [comment]
Gotcha; thanks for maintaining this repo. Super helpful for me.
